app/dubboctl/internal/manifest/render/render.go

```diff
@@ -325,8 +325,10 @@ func renderManifest(valsYaml string, cht *chart.Chart, builtIn bool, opts *Rende
 		if file == "" {
 			continue
 		}
+		// the formatted yaml file should terminate with a newline separator
 		if !strings.HasSuffix(file, YAMLSeparator) {
```

Comment:
Edge case:

```yaml
apiVersion: dubbo.apache.org/v1alpha1
kind: ConditionRoute
metadata:
  name: admin-cp-conditionroute
  namespace: dubbo-system
spec:
  enabled: true
  force: true
  scope: service
  runtime: true
  priority: 100
  key: org.apache.dubbo.samples.CommentService
  configVersion: v3.0
  conditions:
    - method=getComment => region=Hangzhou
--- # there is no EOF (that is `\n` here)
```

The `if` condition will never be true, as the `util.SpaceFilter()` will not preserve the EOF line. Adding a new filter `ReserveEoFSpaceFilter` to resolve it.